Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,162 @@ extension HTTP2ServerTransport.Config {
+      }
+
+      /// The certificate will be provided as an array of bytes.
+      public static func certificate(
+        bytes: [UInt8],
+        serializationFormat: SerializationFormat
+      ) -> Self {
+        Self(wrapped: .certificate(bytes: bytes, serializationFormat: serializationFormat))
+      }
+
+      /// The file path to the location of the certificate, if the source was set to ``file(path:serializationFormat:)``.
+      public var filePath: String? {
+        switch wrapped {
+        case .file(let path, _):
+          return path
+        case .certificate:
+          return nil
+        }
+      }
+
+      /// The bytes of the certificate, if the source was set to ``certificate(bytes:serializationFormat:)``.
+      public var certificateBytes: [UInt8]? {
+        switch wrapped {
+        case .certificate(let bytes, _):
+          return bytes
+        case .file:
+          return nil
+        }
+      }
+
+      /// The serialization format of the certificate.
+      public var serializationFormat: SerializationFormat {
+        switch wrapped {
+        case .file(_, let format):
+          return format
+        case .certificate(_, let format):
+          return format
+        }
+      }
+    }
+
+    public struct PrivateKeySource: Sendable {
+      private enum Wrapped {
+        case file(path: String, serializationFormat: SerializationFormat)
+        case privateKey(bytes: [UInt8], serializationFormat: SerializationFormat)
+      }
+
+      private let wrapped: Wrapped
+
+      /// The private key will be provided via a file.
+      public static func file(path: String, serializationFormat: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, serializationFormat: serializationFormat))
+      }
+
+      /// The private key will be provided as an array of bytes.
+      public static func privateKey(
+        bytes: [UInt8],
+        serializationFormat: SerializationFormat
+      ) -> Self {
+        Self(wrapped: .privateKey(bytes: bytes, serializationFormat: serializationFormat))
+      }
+
+      /// The file path to the location of the private key, if the source was set to ``file(path:serializationFormat:)``.
+      public var filePath: String? {
+        switch wrapped {
+        case .file(let path, _):
+          return path
+        case .privateKey:
+          return nil
+        }
+      }
+
+      /// The bytes of the private key, if the source was set to ``privateKey(bytes:serializationFormat:)``.
+      public var privateKeyBytes: [UInt8]? {
+        switch wrapped {
+        case .privateKey(let bytes, _):
+          return bytes
+        case .file:
+          return nil
+        }
+      }
+
+      /// The serialization format of the private key.
+      public var serializationFormat: SerializationFormat {
+        switch wrapped {
+        case .file(_, let format):
+          return format
+        case .privateKey(_, let format):
+          return format
+        }
+      }
```

Comment:
Yeah probably not, I basically added them because I didn't think of making the `wrapped` property `package` instead of `private`, and I thought I'd may as well have these `public` in case they were useful. But I will just get rid of them.